### PR TITLE
Added provider_proxy for pip

### PR DESCRIPTION
**Pull request by nurriz**

### CODE REVIEW

File: manifests/plugin/rabbitmq.pp [hunk 1]
```diff
@@ -64,10 +65,17 @@
 
   $_manage_package = pick($manage_package, $::collectd::manage_package)
 
-  if $_manage_package {
+  if ($_manage_package) and ($provider_proxy) {
```
Comment (bastelfreak) [inline manifests/plugin/rabbitmq.pp]: can you here only set $install_options to `[{'--proxy' => $provider_proxy}]` and in the else part to `undef`?

This allows us to only define the package resource once.

Comment (nurriz) [inline manifests/plugin/rabbitmq.pp]: Is this what you were thinking?

**Comment by bastelfreak** [comment]
yep, thats what I meant. Can you please add tests as well?

File: manifests/plugin/rabbitmq.pp [hunk 2]
```diff
@@ -64,12 +65,17 @@
 
   $_manage_package = pick($manage_package, $::collectd::manage_package)
 
-  if $_manage_package {
-    package { $package_name:
-      ensure   => $ensure,
-      provider => $package_provider,
-    }
+  if ($_manage_package) and ($provider_proxy) {
```
Comment (jyaworski) [inline manifests/plugin/rabbitmq.pp]: Why are these in parentheses?

Comment (bastelfreak) [inline manifests/plugin/rabbitmq.pp]: they aren't needed, but I'm fine with them.

**Pull request merged**